plugins/uptimerobot/src/components/UptimeHistory/UptimeHistory.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BackstageTheme } from '@backstage/theme';
+import { Bar } from './Bar';
+import { makeStyles } from '@material-ui/core';
+import React from 'react';
+import { Percentage } from '../Percentage';
+import { getUptimeRanges } from '../../utils';
+
+const useStyles = makeStyles<BackstageTheme>(() => ({
+  root: {
+    display: 'flex',
+  },
+}));
+
+export const UptimeHistory = ({
+  data,
+  percentage,
+}: {
+  data: number[];
+  percentage: number;
+}) => {
+  const reversedData = data.reverse();
```

Comment:
The reverse method actually reverses the array _in place_, so it destroys the input!

You're better off just going through the `for` loop backwards instead.